pkg/apis/stork/v1alpha1/backuplocation.go

```diff
@@ -44,6 +45,21 @@ type BackupLocationItem struct {
 	RepositoryPassword string        `json:"repositoryPassword"`
 }
 
+// ClusterItem is the spec used to store a the credentials associated with the cluster
+// Only one of AWSClusterConfig, AzureClusterConfig or GCPClusterConfig should be specified and
+// should match the Type field. Members of the config can be specified inline or
+// through the SecretConfig
+type ClusterItem struct {
```

Comment:
why we need separate `ClusterItem`, can't we add ClusterType to `backuplocationItem` itself. Both share completely identical field

Reply:
Backuplocation's credentials and the cluster's attached credential can be different sometimes.
For ex: Admin can create a backuplocation using his/her creds and share that with a user and the user can add the cluster with his/her creds.

Reply:
Made the changes where `ClusterItem` uses existing cred structures.